### ISSUE TITLE
[Flaky Test] Numerical gradient test for `greater` op

### ISSUE THREAD

**Issue by MasterJH5574** [mlc-ai/relax#135]
**Error log.** https://ci.mlc.ai/blue/organizations/jenkins/mlc-ai-relax/detail/relax/45/pipeline

```
=================================== FAILURES ===================================
___________________ test_binary[greater-relax.greater-llvm] ____________________

target = 'llvm', dev = cpu(0)
binary_op_func = <function greater at 0x7f6df80c5cb0>
binary_op_name = 'relax.greater'

    @tvm.testing.parametrize_targets("llvm")
    def test_binary(target, dev, binary_op_func, binary_op_name):
        data1_numpy = np.random.uniform(1, 2, (3, 3)).astype(np.float32)
        data2_numpy = np.random.uniform(1, 2, (3, 3)).astype(np.float32)
        relax_check_gradients(
>           binary_op_func, binary_op_name, [data1_numpy, data2_numpy], target, dev, (3, 3)
        )

tests/python/relax/test_op_gradient_numeric.py:156: 

...

            if dist > atol * sqrt_n + rtol * grad_norm:
                raise AssertionError(
                    "Analytical and numerical grads wrt '{}' differ too much\n"
                    "analytical grad = {}\n numerical grad = {}\n"
                    "{}% of elements differ, first 10 of wrong positions: {}\n"
                    "distance > atol*sqrt(n) + rtol*grad_norm\n"
                    "distance {} > {}*{} + {}*{}".format(
                        x_name,
                        grad,
                        ngrad,
                        wrong_percentage,
                        wrong_positions[:10],
                        dist,
                        atol,
                        sqrt_n,
                        rtol,
>                       grad_norm,
                    )
                )
E               AssertionError: Analytical and numerical grads wrt '1' differ too much
E               analytical grad = [[0. 0. 0.]
E                [0. 0. 0.]
E                [0. 0. 0.]]
E                numerical grad = [[   0.        0.     -943.2258]
E                [   0.        0.        0.    ]
E                [   0.        0.        0.    ]]
E               11% of elements differ, first 10 of wrong positions: [(0, 2)]
E               distance > atol*sqrt(n) + rtol*grad_norm
E               distance 943.225830078125 > 0.01*3.0 + 0.1*943.225830078125

python/tvm/testing/utils.py:263: AssertionError
=========================== short test summary info ============================
FAILED tests/python/relax/test_op_gradient_numeric.py::test_binary[greater-relax.greater-llvm]
```

**Comment by MasterJH5574** [comment]
cc @Ubospica @SiriusNEO

**Comment by SiriusNEO** [comment]
Fix patch: https://github.com/mlc-ai/relax/pull/136

**Comment by Ubospica** [comment]
Fixed in #136
